src/datepicker/bootstrap-datepicker.component.js

```diff
@@ -0,0 +1,27 @@
+angular.module('patternfly.datepicker').component('pfBootstrapDatepicker', {
+  bindings: {
+    format: '@',
+    date: '<',
+    dateOptions: '@',
```

Comment:
`dateOptions` is an object, not a string, so it needs to be defined as `dateOptions: '<'`.   You'll notice when you make this change, your overrides in your example no longer work Ex:
```
$scope.dateOptions = {
        showWeeks : true
      };
```
:-(

This is because the `$onInit()` is actually called after the objects from the example are bound.   Try adding this to your component:
```
    ctrl.$onChanges = function (changesObj) {
      console.log("on changes");
    };
```
Then in the browser, source tab, open up angular-patternfly.js, find the component, set breakpoints in $onInit() and $onChanges().  You may want to remove the first datepicker example, just to see how the lifecycle of one example works.   You'll see that $onChanges() is called first with the objects from the example, then $onInit() is called afterwards  -in this case overriding the `$scope.dateOptions` set by the user/example.  

We typically have some sort of `if (!angular.isDefined(...)..`  or 'defaulting' operation in the $onInit() to prevent overriding.

Here's some [code](https://github.com/patternfly/angular-patternfly/blob/master/src/toolbars/toolbar-component.js#L15) which has some examples of $onInit(), $onChanges, and even $doCheck for deep comparisions.

Here is another [code example](https://github.com/patternfly/angular-patternfly/blob/master/src/table/tableview/table-view.component.js#L120) where we use lodash `._defaults(...)` to handle default props and user defined props.

Reply:
^^ I'd probably stick with the last example and lodash `._defaults(...)` it's a nice solution.